api/common_test.go

```diff
@@ -0,0 +1,175 @@
+		&ports{{
+			IP:          "0.0.0.0",
+			PrivatePort: 9988,
+			Type:        "tcp",
+		}}: "0.0.0.0:0->9988/tcp",
+		&ports{{
+			PrivatePort: 9988,
+			PublicPort:  8899,
+			Type:        "tcp",
+		}}: "9988/tcp",
+		&ports{{
+			IP:          "4.3.2.1",
+			PrivatePort: 9988,
+			PublicPort:  8899,
+			Type:        "tcp",
+		}}: "4.3.2.1:8899->9988/tcp",
+		&ports{{
+			IP:          "4.3.2.1",
+			PrivatePort: 9988,
+			PublicPort:  9988,
+			Type:        "tcp",
+		}}: "4.3.2.1:9988->9988/tcp",
+		&ports{{
+			PrivatePort: 9988,
+			Type:        "udp",
+		}, {
+			PrivatePort: 9988,
+			Type:        "udp",
+		}}: "9988/udp, 9988/udp",
+		&ports{{
+			IP:          "1.2.3.4",
+			PublicPort:  9998,
+			PrivatePort: 9998,
+			Type:        "udp",
+		}, {
+			IP:          "1.2.3.4",
+			PublicPort:  9999,
+			PrivatePort: 9999,
+			Type:        "udp",
+		}}: "1.2.3.4:9998-9999->9998-9999/udp",
+		&ports{{
+			IP:          "1.2.3.4",
+			PublicPort:  8887,
+			PrivatePort: 9998,
+			Type:        "udp",
+		}, {
+			IP:          "1.2.3.4",
+			PublicPort:  8888,
+			PrivatePort: 9999,
+			Type:        "udp",
+		}}: "1.2.3.4:8887->9998/udp, 1.2.3.4:8888->9999/udp",
+		&ports{{
+			PrivatePort: 9998,
+			Type:        "udp",
+		}, {
+			PrivatePort: 9999,
+			Type:        "udp",
+		}}: "9998-9999/udp",
+		&ports{{
+			IP:          "1.2.3.4",
+			PrivatePort: 6677,
+			PublicPort:  7766,
+			Type:        "tcp",
+		}, {
+			PrivatePort: 9988,
+			PublicPort:  8899,
+			Type:        "udp",
+		}}: "9988/udp, 1.2.3.4:7766->6677/tcp",
+		&ports{{
+			IP:          "1.2.3.4",
+			PrivatePort: 9988,
+			PublicPort:  8899,
+			Type:        "udp",
+		}, {
+			IP:          "1.2.3.4",
+			PrivatePort: 9988,
+			PublicPort:  8899,
+			Type:        "tcp",
+		}, {
+			IP:          "4.3.2.1",
+			PrivatePort: 2233,
+			PublicPort:  3322,
+			Type:        "tcp",
+		}}: "4.3.2.1:3322->2233/tcp, 1.2.3.4:8899->9988/udp, 1.2.3.4:8899->9988/tcp",
+		&ports{{
+			PrivatePort: 9988,
+			PublicPort:  8899,
+			Type:        "udp",
+		}, {
+			IP:          "1.2.3.4",
+			PrivatePort: 6677,
+			PublicPort:  7766,
+			Type:        "tcp",
+		}, {
+			IP:          "4.3.2.1",
+			PrivatePort: 2233,
+			PublicPort:  3322,
+			Type:        "tcp",
+		}}: "9988/udp, 4.3.2.1:3322->2233/tcp, 1.2.3.4:7766->6677/tcp",
+	}
+
+	for valid, expected := range cases {
+		actual := DisplayablePorts(*valid)
+		if expected != actual {
+			t.Fatalf("Expected %s, got %s.", expected, actual)
+		}
+	}
+}
+
+// MatchesContentType
+func TestJsonContentType(t *testing.T) {
+	if !MatchesContentType("application/json", "application/json") {
+		t.Fail()
+	}
+
+	if !MatchesContentType("application/json; charset=utf-8", "application/json") {
+		t.Fail()
+	}
+
+	if MatchesContentType("dockerapplication/json", "application/json") {
+		t.Fail()
+	}
+}
+
+// LoadOrCreateTrustKey
+func TestLoadOrCreateTrustKeyInvalidKeyFile(t *testing.T) {
+	tmpKeyFolderPath, _ := ioutil.TempDir("", "api-trustkey-test")
```

Comment:
Need to defer removal of this, and make sure to check error in case it couldn't create.